packages/salesforcedx-vscode-core/src/commands/forceSourceDeploySourcePath.ts

```diff
@@ -54,6 +57,21 @@ export class MultipleSourcePathsGatherer implements ParametersGatherer<string> {
 }
 
 export async function forceSourceDeploySourcePath(sourceUri: vscode.Uri) {
+  if (!sourceUri) {
+    const editor = vscode.window.activeTextEditor;
+    if (editor && editor.document.languageId !== 'forcesourcemanifest') {
```

Comment:
Only the commands that deal with manifest files should have a check for the language id being forcesourcemanifest e.g. `forceSourceDeployManifest.ts`

Reply:
Update: synced with Jim and we'll keep this condition as it gives a better experience/error message than not having it.